docs/the-new-architecture/create-module-library.md

```diff
@@ -0,0 +1,134 @@
+
+```json
+  "codegenConfig": {
+    "name": "RN<your module name>Spec",
+    "type": "all",
+    "jsSrcsDir": "src",
+    "outputDir": {
+      "ios": "ios/generated",
+      "android": "android/generated"
+    },
+    "android": {
+      "javaPackageName": "com.<name-of-the-module>"
+    }
+  },
+```
+
+Finally the library contains already all the infrastruction to let the library be linked with iOS and Android.
+
+### 2. Copy the Code over from Your App
+
+The rest of the guide assumes that you have a local Turbo Native Module in your app, created following the guidelines shown in the other guides in the website: platform specific Turbo Native Modules; [cross-platform Turbo Native Modules](/docs/the-new-architecture/pure-cxx-modules.md).
+
+<!-- TODO: add links for Turbo Native Modules -->
+
+1. Move the code you have in the `specs` folder in your app into the `src` folder created by the `create-react-native-library` folder.
+2. Update the `index.ts` file to properly export the Turbo Native Module spec so that it is accessible from the library. For example:
+
+```ts
+import NativeSampleModule from './NativeSampleModule';
+
+export default NativeSampleModule;
+```
+
+3. Copy the native module over:
+
+   - Replace the code in the `android/src/main/java/com/<name-of-the-module>` with the code you wrote in the app for your native module, if any.
+   - Replace the code in the `ios` folder with the code you wrote in your app for your native module, if any.
+   - Replace the code in the `cpp` folder with the code you wrote in your app for your native module, if any.
+
+4. Update all the references from the previous spec name to the new spec name, the one that is defined in the `codegenConfig` field of the library's `package.json`. For example, if in the app `package.json` you set `AppSpecs` as `codegenConfig.name` and in the library it is called `RNNativeSampleModuleSpec`, you have to replace every occurrence of `AppSpecs` with `RNNativeSampleModuleSpec`.
+
+That's it! You have moved all the required code out of your app and in a separate library.
+
+## Testing your Library
+
+The `create-react-native-library` comes with an useful example application that is already configured to work properly with the library. This is a great way to test it!
+
+If you look at the `example` folder, you can find the same structure of a new React Native application that you can create from the [`react-native-community/template`](https://github.com/react-native-community/template).
+
+To test your library:
+
+1. Navigate to the `example` folder.
+2. Run `yarn install` to install all the dependencies.
+3. For iOS only, you need to install CocoaPods: `cd ios && pod install`.
+4. Build and run Android with `yarn android` from the `example` folder.
+5. Build and run iOS with `yarn ios` from the `example` folder.
+
```

Comment:
I wanted to add a paragraph on the Local Module setup, but it is currently broken. Metro can't follow the symlinks that yarn generates and it can't pick up the JS files from the other folder.

Probably there is a way to make it work by tinkering with the `metro.config.js`, but I don't think we should go to that level of details in these guides, so I decided to leave that setup out for simplicity. There is a way to test the library anyway.

cc. @huntie if you know how to handle this case in a simple way.